dynatrace/dynatrace_test.go

```diff
@@ -0,0 +1,180 @@
+	"go.opentelemetry.io/otel/sdk/export/metric/aggregation"
+	"go.uber.org/zap"
+)
+
+func TestNewExporter(t *testing.T) {
+	t.Run("construct with URL", func(t *testing.T) {
+		got, err := NewExporter(Options{URL: "https://example.com"})
+		if err != nil {
+			t.Error("Should not return error")
+		}
+		if got.opts.URL != "https://example.com" {
+			t.Errorf("Expected %v but got %v", "https://example.com", got.opts.URL)
+		}
+	})
+
+	t.Run("create a logger when missing", func(t *testing.T) {
+		got, err := NewExporter(Options{})
+		if err != nil {
+			t.Error("Should not return error")
+		}
+		if got.logger == nil {
+			t.Error("Exporter missing logger")
+		}
+	})
+
+	t.Run("create a metric name formatter when missing", func(t *testing.T) {
+		got, err := NewExporter(Options{})
+		if err != nil {
+			t.Error("Should not return error")
+		}
+		if got.opts.MetricNameFormatter == nil {
+			t.Error("Exporter missing metric name formatter")
+		}
+	})
+
+	t.Run("use default url when missing", func(t *testing.T) {
+		got, err := NewExporter(Options{})
+		if err != nil {
+			t.Error("Should not return error")
+		}
+		if got.opts.URL != apiconstants.GetDefaultOneAgentEndpoint() {
+			t.Errorf("Expected %v but got %v", apiconstants.GetDefaultOneAgentEndpoint(), got.opts.URL)
+		}
+	})
+}
+
+func Test_defaultFormatter(t *testing.T) {
+	type args struct {
+		namespace string
+		name      string
+	}
+	tests := []struct {
+		name string
+		args args
+		want string
+	}{
+		{
+			"returns name",
+			args{"namespace", "name"},
+			"name",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := defaultFormatter(tt.args.namespace, tt.args.name); got != tt.want {
+				t.Errorf("defaultFormatter() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestExporter_ExportKindFor(t *testing.T) {
+	type fields struct {
+		opts              Options
+		defaultDimensions dimensions.NormalizedDimensionList
+		staticDimensions  dimensions.NormalizedDimensionList
+		client            *http.Client
+		logger            *zap.Logger
+	}
+	type args struct {
+		in0 *metric.Descriptor
+		in1 aggregation.Kind
+	}
+	tests := []struct {
+		name   string
+		fields fields
+		args   args
+		want   export.ExportKind
+	}{
+		{
+			"returns delta",
+			fields{},
+			args{&metric.Descriptor{}, aggregation.ExactKind},
+			export.DeltaExporter,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			e := &Exporter{
+				opts:              tt.fields.opts,
+				defaultDimensions: tt.fields.defaultDimensions,
+				staticDimensions:  tt.fields.staticDimensions,
+				client:            tt.fields.client,
+				logger:            tt.fields.logger,
+			}
+			if got := e.ExportKindFor(tt.args.in0, tt.args.in1); !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("Exporter.ExportKindFor() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestExporter_send(t *testing.T) {
+	t.Run("authenticates requests", func(t *testing.T) {
```

Comment:
Does this test case test if the Authorization header of the request indeed contains a `Api-Token token`?

Reply:
Line 143